tutorials/financial-tick-data/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,438 @@
+---
+title: Ingest real-time financial websocket data
+excerpt: Set up a data pipeline to get data from different financial APIs
+products: [cloud, mst, self_hosted]
+keywords: [finance, analytics, websockets, data pipeline]
+---
+
+import CreateHypertableStocks from "versionContent/_partials/_create-hypertable-twelvedata-stocks.mdx";
+import GraphOhlcv from "versionContent/_partials/_graphing-ohlcv-data.mdx";
+
+# Ingest real-time financial websocket data
+
+This tutorial shows you how to ingest real-time time-series data into
+TimescaleDB using a websocket connection. The tutorial sets up a data pipeline to
+ingest real-time data from our data partner, [Twelve Data][twelve-data].
+Twelve Data provides a number of different financial APIs, including stock,
+crypto, forex, ETFs, and more. It also supports websocket connections in case
+you want to update your database frequently. With websockets, you need to
+connect to the server, subscribe to symbols, and you can start receiving data
+in real-time during market hours.
+
+When you complete this tutorial, you'll have a data pipeline set
+up that ingests real-time financial data into your Timescale.
+
+This tutorial uses Python and the API
+[wrapper library][twelve-wrapper] provided by Twelve Data.
+
+## Prerequisites
+
+Before you begin, make sure you have:
+
+*   A Timescale [service and connect to the service][financial-tick-dataset].
+*   Download the `.sql` file that contains the credentials such as `<HOST>`,
+    `<PORT>`, `<PASSWORD>`, and other details to connect to Timescale
+    service. You can also find these details in the `Connection Info` section of
+    your Timescale service in the Timescale portal.
+*   Installed Python 3
+*   Signed up for [Twelve Data][twelve-signup]. The free tier is perfect for
+    this tutorial.
+*   Made a note of your Twelve Data [API key](https://twelvedata.com/account/api-keys).
+
+<Collapsible heading="Connect to the websocket server">
+
+When you connect to the Twelve Data API through a websocket, you create a
+persistent connection between your computer and the websocket server. You need
+to first set up a Python environment and then pass two arguments to create a
+websocket object and establish connection.
+
+## Set up a new Python environment
+
+Create a new Python virtual environment for this project and activate it. All
+the packages you need to complete for this tutorial are installed in this environment.
```

Comment:
```suggestion
Start by creating a new Python virtual environment for this project and activate it. All
the packages you need for this tutorial are installed in this environment.
```

Reply:
less wordy sentences and be imperative again style guide.